source/simulator/initial_conditions.cc

```diff
@@ -174,6 +174,104 @@ namespace aspect
 
 
   template <int dim>
+  void Simulator<dim>::interpolate_particle_properties (const AdvectionField &advection_field)
+  {
+    // below, we would want to call VectorTools::interpolate on the
+    // entire FESystem. there currently is no way to restrict the
+    // interpolation operations to only a subset of vector
+    // components (oversight in deal.II?), specifically to the
+    // temperature component. this causes more work than necessary
+    // but worse yet, it doesn't work for the DGP(q) pressure element
+    // if we use a locally conservative formulation since there the
+    // pressure element is non-interpolating (we get an exception
+    // even though we are, strictly speaking, not interested in
+    // interpolating the pressure; but, as mentioned, there is no way
+    // to tell VectorTools::interpolate that)
```

Comment:
Hm. I wonder if we could somehow extract that code and put it into a common function, rather than duplicate code. Alternatively, we may also extend the deal.II function and call that when available -- in that case, the code duplication would eventually go away as we require a newer deal.II version.